configLoader.js

```diff
@@ -0,0 +1,438 @@
+              }
+              if (content.customLink) {
+                finalContent += content.customLink;
+              }
+
+              popupTemplateMap.set(id, { title, content: finalContent });
+            }
+          });
+        }
+        config["popupTemplateMap"] = popupTemplateMap;
+        config["data"] = data;
+
+        resolve(config);
+      })
+      .catch((err) => console.error(err));
+  });
+}
+
+/*
+ * Purify html content
+ * Only p a[href|target] strong em ul ol li div[class] are allowed
+ */
+function purifyContent(content) {
+  return DOMPurify.sanitize(content, {
+    ALLOWED_TAGS: ["p", "a", "strong", "em", "ul", "ol", "li", "div"],
+    ALLOWED_ATTR: ["href", "target", "class"],
+  });
+}
+
+/**
+ *  Build popup title and content from global configurations
+ *  Content have three parts : customContent , defaultTable , customLink
+ * @param {*} feature
+ * @param {*} config
+ * @returns
+ */
+function buildDefaultPopup(feature, config) {
+  let title = feature.properties.name;
+  let content = {};
+
+  if (config.popupTitle) {
+    const matches = config.popupTitle.match(/{(.*?)}/g);
+    const variablesExist = matches
+      ? matches.every((match) =>
+          feature.properties.hasOwnProperty(match.slice(1, -1))
+        )
+      : true;
+
+    // If all variables exist, use the custom title, otherwise use the name
+    if (variablesExist) {
+      title = config.popupTitle.replace(
+        /{(.*?)}/g,
+        (_, key) => feature.properties[key]
+      );
+    }
+  }
+
+  if (config.popupContent) {
+    const matches = config.popupContent.match(/{(.*?)}/g);
+    const variablesExist = matches
+      ? matches.every((match) =>
+          feature.properties.hasOwnProperty(match.slice(1, -1))
+        )
+      : true;
+
+    if (variablesExist) {
+      let res = purifyContent(
+        config.popupContent.replace(
+          /{(.*?)}/g,
+          (_, key) => feature.properties[key]
+        )
+      );
+
+      if (res && res != "") {
+        content.customContent = res; // Purify the content to prevent XSS
+      }
+    }
+  }
+
+  //Default table content
+  content.defaultTable = buildPopupContentTable(
+    feature,
+    config.popupFieldLabels,
+    config.popupAllowedFields,
+    config.popupBlockedFields
+  );
+
+  if (config.popupLinks) {
+    let links = "";
+    config.popupLinks.forEach((link) => {
+      links += `<p><a href="${link.link}" target="${
+        link.target ? link.target : "_blank"
+      }">${link.text}</a></p>`;
+    });
+    content.customLink = links;
+  }
+
+  return { title, content };
+}
+
+/**
+ * Build defaultTable for popup content
+ * @param {*} feature  GeoJSON feature for each pin
+ * @param {*} fieldLabels Custom field label display . HashMap
+ * @param {*} allowedFields White list for fields to display . HashSet
+ * @param {*} blockedFields Black list for fields to display . HashSet
+ * @param {*} links Links to display . Array
```

Comment:
For all comments similar to this, please include the specific types of parameters in the comments for functions. Should be it more explicit such as {string}, {Object}, {Array}, {Classname}.